### PR TITLE
[nextjs] Update feaas clientside and adjust types in byoc bundlers

### PR DESCRIPTION
**Pull request by art-alexeyenko**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [ ] Changelog updated

## Description / Motivation
1) Updated sitecore-feaas dependency to latests to accomodate:
- ContextID changes
- New types for bundler
2) Adjusted comments
3) Adjusted BYOC import-export logic with @Inviz 's suggestions (ground work for AppRouter BYOC support)
4) Also locked the dependency version in sitecore-jss-react package to avoid breaking changes breaking the package.

## Testing Details
- [ ] Unit Test Added
- [x] Manual Test/Other (Please elaborate)

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.ts [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 const ClientBundle = dynamic(() => import('./index.client'), {
   ssr: false,
 });
+FEAAS.enableNextClientsideComponents(dynamic, ClientBundle);
 
-// Import your SSR/hybrid components via index.hybrid.ts
+// Import your hybrid (server rendering with client hydration) components via index.hybrid.ts
 import './index.hybrid';
 
 // As long as client bundle is exported and rendered on page (as an empty element), client-only BYOC components are registered and become available
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.ts:18]: Does this comment need some updating with the change? There's a bit of magic now between the `FEAAS.enableNextClientsideComponents` above and `FEAAS.ExternalComponentBundle` below.

File: packages/sitecore-jss-react/package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
     "url": "https://github.com/sitecore/jss/issues"
   },
   "devDependencies": {
-    "@sitecore-feaas/clientside": "^0.4.4",
+    "@sitecore-feaas/clientside": "0.4.12",
```
Comment (ambrauer) [inline packages/sitecore-jss-react/package.json:30]: Why the exact version here? Should at least match our peerDependencies (~). 
In general though, are we anticipating breaking changes in @sitecore-feaas/clientside minor versions? If following semver (which we should be), then ^ should be fine.

Comment (Inviz) [reply packages/sitecore-jss-react/package.json:30]: yeah needs ^ probably, if it was copied from my app - sorry

File: packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.ts (hunk 1 above)
Comment (art-alexeyenko) [reply packages/create-sitecore-jss/src/templates/nextjs/src/byoc/index.ts:18]: Hybrid import is unaffected - but it makes sense to update the client import comments, or move the enableClientside call

File: packages/sitecore-jss-react/package.json (hunk 2 above)
Comment (art-alexeyenko) [reply packages/sitecore-jss-react/package.json:30]: > In general though, are we anticipating breaking changes in @sitecore-feaas/clientside minor versions

We don't, but sometimes they happen. I've encountered two instances during Forms integration dev. I can bring the ^ back if we don't expect any other ones.
@ambrauer

Comment (art-alexeyenko) [reply packages/sitecore-jss-react/package.json]: Updated the version

**Review by ambrauer — APPROVED** [review]
Looks good to me

**Pull request merged**